bot/exts/events/hacktoberfest/_utils.py

```diff
@@ -0,0 +1,451 @@
+import logging
+import random
+import re
+from collections import Counter
+from datetime import datetime, timedelta
+from typing import Optional, Union
+from urllib.parse import quote_plus
+
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours, NEGATIVE_REPLIES, Tokens
+
+log = logging.getLogger(__name__)
+
+# Constants for `.hacktoberfest issue`
+ISSUES_REQUEST_HEADERS = {
+    "User-Agent": "Python Discord Hacktoberbot",
+    "Accept": "application / vnd.github.v3 + json"
+}
+if GITHUB_TOKEN := Tokens.github:
+    ISSUES_REQUEST_HEADERS["Authorization"] = f"token {GITHUB_TOKEN}"
+
+# Constants for `.hacktoberfest stats`
+CURRENT_YEAR = datetime.now().year  # Used to construct GH API query
+PRS_FOR_SHIRT = 4  # Minimum number of PRs before a shirt is awarded
+REVIEW_DAYS = 14  # number of days needed after PR can be mature
+
+STATS_REQUEST_HEADERS = {"User-Agent": "Python Discord Hacktoberbot"}
+GITHUB_TOPICS_ACCEPT_HEADER = {"Accept": "application/vnd.github.mercy-preview+json"}
+
+# using repo topics API during preview period requires an accept header
+if GITHUB_TOKEN := Tokens.github:
+    STATS_REQUEST_HEADERS["Authorization"] = f"token {GITHUB_TOKEN}"
+    GITHUB_TOPICS_ACCEPT_HEADER["Authorization"] = f"token {GITHUB_TOKEN}"
+
+GITHUB_NONEXISTENT_USER_MESSAGE = (
+    "The listed users cannot be searched either because the users do not exist "
+    "or you do not have permission to view the users."
+)
+
+URL = (
+    "https://api.github.com/search/issues?"  # base url
+    "per_page=100"  # limit results per-page returned by API to 100 (the maximum)
+    "&q="  # add query parameters
+    "is:issue+"  # is an issue...
+    "state:open+"  # ...that's open...
+    "label:hacktoberfest+"  # ...with the `hacktoberfest` label...
+    "language:python"  # ...and in Python.
```

Comment:
```suggestion
    "https://api.github.com/search/issues?"  # base url
    "per_page=100"                           # limit results per-page returned by API to 100 (the maximum)
    "&q="                                    # add query parameters
    "is:issue+"                              # is an issue
    "state:open+"                            # that's open
    "label:hacktoberfest+"                   # has `hacktoberfest` label
    "language:python"                        # in Python
```
Not sure if the width would pass lint test

Reply:
The width does pass and I like it, so have done this :+1:

I'll also remove the `...` since I think it adds more noise rather than helping with flow as intended.